### ISSUE TITLE
E2e tests fail locally, in somewhat stochastic ways

### ISSUE THREAD

**Issue by anna-parker** [loculus-project/loculus#2713]
I run the following set up steps locally:

1. Set up local pods:
    ```
    ../deploy.py cluster --dev
    ../deploy.py helm --dev
    ```
3. Run backend locally in separate terminal:
    ```
    ../generate_local_test_config.sh
    ./start_dev.sh
    ```
4. Run webpage locally in separate terminal:
    ```
    npm ci
    npx playwright install
    npm run dev
    ```
5. Run tests in separate terminal
    ```
    npm run e2e
    ```

<img width="841" alt="image" src="https://github.com/user-attachments/assets/13db54e4-4d47-434c-af25-467ab823b318">

For example
```
7) [chromium] › pages/seqsets/[seqSetId].spec.ts:84:5 › The seqSet item page › edit functionality updates seqSet and increases version 

    Retry #1 ───────────────────────────────────────────────────────────────────────────────────────

    Error: [error]: 'Warning: Encountered two children with the same key, `%s`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.%s LOC_SS_2 

       at util/throwOnConsole.ts:24

      22 |             return;
      23 |         }
    > 24 |         throw new Error(`[${message.type()}]: '${message.text()}'`);
         |               ^
      25 |     };
      26 |
      27 |     page.on('console', listener);

        at tbody
        at table
        at div
        at SeqSetList (http://localhost:3000/src/components/SeqSetCitations/SeqSetList.tsx:104:30)'
        at Page.listener (/Users/aparker/Documents/Github/loculus/website/tests/util/throwOnConsole.ts:24:15)

    Error: locator.click: Test ended.
    Call log:
      - waiting for getByRole('button', { name: 'Confirm' })
      -   locator resolved to <button class="btn loculusColor text-white hover:bg-primary-700">Confirm</button>
      - attempting click action
      -   waiting for element to be visible, enabled and stable
```

**Comment by theosanderson** [comment]
If we want to ensure this doesn't happen we could add a test that works like this

**Comment by corneliusroemer** [comment]
@theosanderson - what do you mean exactly? You think this might happen because of the dev setup being different than how we currently run e2e tests in CI?

I could reproduce these using the exact steps of Anya (I added the missing ones).

It's possible that the bugs happen only locally with these steps due to a) website being in dev mode, b) backend being run locally rather than in kubernetes, c) the test failures being due to race conditions that require certain timing - a slower website might trigger them for example, but in CI playwright might be slower than the website so it doesn't get triggered.

I fixed a bunch of flakes in the past - the tests were not expecting the right thing. This wasn't an issue in CI, only locally. I only fixed the tests that failed locally for me though, and my setup was likely different than here. So maybe running them differently might help, and me using this setup here and deflaking the ones that still fail.

```
  Slow test file: [chromium] › pages/sequences/accession.spec.ts (32.0s)
  Consider splitting slow test files to speed up parallel execution
  3 failed
    [chromium] › pages/review/index.spec.ts:22:5 › The review page › should allow bulk approval ────
    [chromium] › pages/review/index.spec.ts:38:5 › The review page › should allow bulk deletion ────
    [chromium] › pages/seqsets/[seqSetId].spec.ts:84:5 › The seqSet item page › edit functionality updates seqSet and increases version
  3 flaky
    [chromium] › pages/search/index.spec.ts:23:5 › The search page › should search one existing sequence entry by accession, then click it
    [chromium] › pages/seqsets/[seqSetId].spec.ts:24:5 › The seqSet item page › displays layout correctly
    [chromium] › pages/seqsets/index.spec.ts:24:5 › The seqSets list page › successfully creates test seqSet in beforeAll
  1 skipped
  29 passed (2.3m)
```

**Comment by corneliusroemer** [comment]
When replacing `npm run dev` (which doesn't make much sense to use for E2E, it's terribly slow, 3m20 instead of 20s) with `npm run start-server` I only get 1 flaky and no failure.

This doesn't mean that there aren't any test issues, just that they only show when one runs E2E tests in an inferior way.

```
      101 |         await searchPage.clickResetButton();
      102 |
    > 103 |         await expect(searchPage.getAccessionField()).toHaveValue('');
          |                                                      ^
      104 |     });
      105 |
      106 |     async function performDownload(page: Page, options: PerformDownloadOptions = {}): Promise<string> {

        at /Users/corneliusromer/code/loculus/website/tests/pages/search/index.spec.ts:103:54

    attachment #1: screenshot (image/png) ──────────────────────────────────────────────────────────
    test-results/pages-search-index-The-search-page-should-reset-the-search-chromium/test-failed-1.png
    ────────────────────────────────────────────────────────────────────────────────────────────────

    attachment #2: trace (application/zip) ─────────────────────────────────────────────────────────
    test-results/pages-search-index-The-search-page-should-reset-the-search-chromium/trace.zip
    Usage:

        npx playwright show-trace test-results/pages-search-index-The-search-page-should-reset-the-search-chromium/trace.zip

    ────────────────────────────────────────────────────────────────────────────────────────────────

  1 flaky
    [chromium] › pages/search/index.spec.ts:90:5 › The search page › should reset the search ───────
  1 skipped
  34 passed (21.6s)
```

**Comment by anna-parker** [comment]
Just checked locally - for me tests fail when I run the website with `npm run start` and `npm run dev` but as @corneliusroemer mentioned `npm run start-server` does not lead to any test failures.

**Comment by theosanderson** [comment]
> @theosanderson - what do you mean exactly? You think this might happen because of the dev setup being different than how we currently run e2e tests in CI?

Yep, I would guess that's why the tests don't fail in CI

**Comment by corneliusroemer** [comment]
The docs recommended the wrong=slow=flaky `npm run serve` = `npm run start`, so I've got a PR to fix this.

The tests likely are still flaky, we're just avoiding the flakes by having a fast enough website. I guess the flakes happen when the website is too slow compared to playwrigh.

**Comment by corneliusroemer** [comment]
I think this is mostly fixed now with #2715
